aperture.go

```diff
@@ -208,15 +211,75 @@ func (a *Aperture) Start(errChan chan error) error {
 		}()
 	}
 
-	// Initialize our etcd client.
-	a.etcdClient, err = clientv3.New(clientv3.Config{
-		Endpoints:   []string{a.cfg.Etcd.Host},
-		DialTimeout: 5 * time.Second,
-		Username:    a.cfg.Etcd.User,
-		Password:    a.cfg.Etcd.Password,
-	})
-	if err != nil {
-		return fmt.Errorf("unable to connect to etcd: %v", err)
+	var (
+		secretStore mint.SecretStore
+		onionStore  tor.OnionStore
+	)
+
+	// Connect to the chosen database backend.
+	switch a.cfg.DatabaseBackend {
+	case "etcd":
+		// Initialize our etcd client.
+		a.etcdClient, err = clientv3.New(clientv3.Config{
+			Endpoints:   []string{a.cfg.Etcd.Host},
+			DialTimeout: 5 * time.Second,
+			Username:    a.cfg.Etcd.User,
+			Password:    a.cfg.Etcd.Password,
+		})
+		if err != nil {
+			return fmt.Errorf("unable to connect to etcd: %v", err)
+		}
+
+		secretStore = newSecretStore(a.etcdClient)
+		onionStore = newOnionStore(a.etcdClient)
+
+	case "postgres":
+		db, err := aperturedb.NewPostgresStore(a.cfg.Postgres)
+		if err != nil {
+			return fmt.Errorf("unable to connect to postgres: %v",
+				err)
+		}
+		a.db = db.DB
+
+		dbSecretTxer := aperturedb.NewTransactionExecutor(db,
+			func(tx *sql.Tx) aperturedb.SecretsDB {
+				return db.WithTx(tx)
+			},
+		)
+		secretStore = aperturedb.NewSecretsStore(dbSecretTxer)
+
+		dbOnionTxer := aperturedb.NewTransactionExecutor(db,
+			func(tx *sql.Tx) aperturedb.OnionDB {
+				return db.WithTx(tx)
+			},
+		)
+		onionStore = aperturedb.NewOnionStore(dbOnionTxer)
+
+	case "sqlite":
+		db, err := aperturedb.NewSqliteStore(a.cfg.Sqlite)
```

Comment:
Looks like if only `databasebackend: sqlite` is set, then the db file isn't set, so it ends up being `""`.

Reply:
fixed